### ISSUE TITLE
Lots of warrnings using gcc 8.2.1 and clang 8 on Arch Linux

### ISSUE THREAD

**Issue by moneroexamples** [Corvusoft/restbed#367]
Using `cmake -DBUILD_SSL=FALSE ..` and `make` on Arch Linux. Most warnings are on gcc. There are far less on clang.

#### gcc (GCC) 8.2.1 20181127
Examples of warnings:

```
std::enable_shared_from_this<restbed::Session>? has accessible non-virtual destructor [-Wnon-virtual-dtor]
     class Session : public std::enable_shared_from_this< Session >
std::enable_shared_from_this<restbed::Session>? has accessible non-virtual destructor [-Wnon-virtual-dtor]
     class Session : public std::enable_shared_from_this< Session >
```

Full `make` output is here:
https://paste.fedoraproject.org/paste/ZKvV68Ux6BZg0GdgTyAtDQ

#### clang version 8.0.0 (tags/RELEASE_800/final)
Examples:

```
/home/mwo2/restbed/source/corvusoft/restbed/detail/socket_impl.cpp:428:23: warning: lambda capture 'this' is not used [-Wunused-lambda-capture]
                    [ this, finished, sharedSize, sharedError ]( const error_code & error, size_t size ) {
                      ^~~~~~
/home/mwo2/restbed/source/corvusoft/restbed/detail/socket_impl.cpp:577:23: warning: lambda capture 'this' is not used [-Wunused-lambda-capture]
                    [ this, finished, sharedLength, sharedError ]( const error_code & error, size_t length ) {
```

Full `make` output:
https://paste.fedoraproject.org/paste/-JHNRD5w7laZNrzVK~dtjw

**Comment by ben-crowhurst** [comment]
Thanks will look to perform a wee clean-up.

**Comment by ben-crowhurst** [comment]
This has now been resolved in the 4.7 release branch.
